opentelemetry-api/src/opentelemetry/_metrics/__init__.py

```diff
@@ -129,17 +130,27 @@ def version(self):
     def schema_url(self):
         return self._schema_url
 
-    # FIXME check that the instrument name has not been used already
+    def _check_instrument_name(self, name, type_, unit, description):
+
+        instrument_id = "".join(
+            [name.strip().lower(), str(type_), unit, description]
+        )
+
+        if instrument_id in self._instrument_ids:
+            _logger.warning("Instrument name %s has been used already", name)
```

Comment:
update this for more accurate message since `name` alone is not the identity?

Reply:
Probably want to rename the function as well.

Reply:
Fixed